surveys/static/js/charts.js

```diff
@@ -0,0 +1,336 @@
+   * objects, see surveys.views.SurveySubmittedDetail.
+   * @constructor
+   * @param {Object} surveys - A data object containing survey results.
+   * @param {Object} types - An object of type definitions for survey results.
+   * @param {Object} bins - An object of bin definitions for binning survey results.
+   */
+  this.surveys = surveys;
+  this.countTypes = types.count;
+  this.observationalTypes = types.observational;
+  this.interceptTypes = types.intercept;
+  this.freeResponseInterceptTypes = types.freeResponseIntercept;
+  this.freeResponseInterceptBins = bins.freeResponseIntercept;
+}
+
+ChartHelper.prototype.loadChart = function(chartId, chartTitle, dataSourceId) {
+  /**
+   * Load a Highcharts chart.
+   * @param {String} chartId - The ID attribute of the chart container on the page.
+   * @param {String} chartTitle - The title of the chart to display.
+   * @param {String} dataSourceId - The ID of the primary data source to display.
+   */
+  // Check the type of the result to determine what kind of chart to display
+  var resultType;
+  if (this.surveys.length === 0) {
+    throw new Error('This survey has no collected data.');
+  } else {
+    resultType = this.surveys[0].data[dataSourceId].type;
+  }
+
+  var isCount = this.countTypes.indexOf(resultType) > -1;
+  var isObservational = this.observationalTypes.indexOf(resultType) > -1;
+  var isIntercept = this.interceptTypes.indexOf(resultType) > -1;
+  var isFreeResponseIntercept = this.freeResponseInterceptTypes.indexOf(resultType) > -1;
+
+  if (!(isCount || isObservational || isIntercept || isFreeResponseIntercept)) {
+    // The data object needs to be one of the valid types
+    throw new Error('Not a valid chart type: ' + resultType);
+  } else {
+    // Retrieve chart data depending on the data source type
+    var chartData = [];
+    var yAxisLabel = (isCount) ? 'Median response' : '% of responses';
+    if (isCount) {
+      chartData = this._getCountChartData(dataSourceId, chartTitle);
+    } else if (isObservational) {
+      chartData = this._getObservationalChartData(dataSourceId);
+    } else if (isIntercept) {
+      chartData = this._getInterceptChartData(dataSourceId);
+    } else {
+      chartData = this._getFreeResponseInterceptChartData(dataSourceId);
+    }
+  }
+  // Format the chart data for Highcharts display
+  var categories = [];
+  var series = [{
+    data: []
+  }];
+  chartData.forEach(function(data) {
+    // chartData elements are stored as an Array where the first element is the
+    // category name and the second element is the value -- a map would be easier
+    // to work with, but wouldn't preserve the order of the categories
+    var category = data[0];
+    var value = data[1];
+    categories.push(category);
+    series[0].data.push(value);
+  });
+  // Initialize a Highcharts chart
+  Highcharts.chart(chartId, {
+    chart: {
+      type: 'column',
+      backgroundColor: '#FFFFFF',
```

Comment:
A bullet from #96: "Select colors randomly from a data viz palette". I don't think that needs to happen in this PR but it'd be great to include in the MVP! Maybe a separate issue?

Reply:
Oops, another spec item I missed! Added this on to #100, which is the point at which we'll actually be introducing multiple data sources (and so will need multiple colors).